### ISSUE TITLE
SyntaxError: Unexpected token (package.json) with Webpack

### ISSUE THREAD

**Issue by debasispanda** [webpack/webpack#2541]
ERROR in Entry module not found: SyntaxError: D:\ProjectPath\package.json (directory description file): SyntaxError: Unexpected token

**webpack.config.js**
---------------------------

```javascript
module.exports = {
    context: __dirname,
    entry: 'main.ts',
    output: {
        path: __dirname + '/Builds/js',
        filename: 'all.js'
    },
    resolve: {
        extensions: ['', '.js', '.ts']
    },
    module: {
        loaders: [
          {
              test: /\.ts$/,
              loader: 'ts'
          },
          {
              test: /\.json$/,
              loader: "json"
          }
        ]
    }
};
```

**package.json**
------------------------------------
```javascript
{
    "version": "1.0.0",
    "name": "RevenueTracker",
    "description": "A angular 2 app",    
    "scripts": {
        "build": "webpack",
        "typings": "typings",
        "preinstall": "npm i -g bower",
        "postinstall": "typings install"
    },
    "dependencies": {
        "jquery": "~1.11.3",
        "@angular/common": "2.0.0-rc.1",
        "@angular/compiler": "2.0.0-rc.1",
        "@angular/core": "2.0.0-rc.1",
        "@angular/http": "2.0.0-rc.1",
        "@angular/platform-browser": "2.0.0-rc.1",
        "@angular/platform-browser-dynamic": "2.0.0-rc.1",
        "@angular/router": "2.0.0-rc.1",
        "@angular/router-deprecated": "2.0.0-rc.1",
        "@angular/upgrade": "2.0.0-rc.1",
        "systemjs": "0.19.26",
        "es6-shim": "^0.35.0",
        "reflect-metadata": "0.1.2",
        "rxjs": "5.0.0-beta.6",
        "zone.js": "^0.6.12",
        "highcharts": "^4.2.3",
        "ng2-highcharts": "^0.3.4",
        "moment": "^2.13.0"
    },
    "devDependencies": {
        "typescript": "^1.8.0",
        "typings": "^0.7.12",
        "gulp": "3.9.0",
        "gulp-bower": "^0.0.13",
        "gulp-uglify": "1.5.1",
        "gulp-concat": "2.6.0",
        "gulp-sourcemaps": "1.6.0",
        "require-dir": "0.3.0",
        "webpack": "^1.12.14",
        "webpack-merge": "^0.9.0",
        "file-loader": "^0.8.5",
        "ts-loader": "^0.8.1",
        "json-loader": "0.5.4"
    }
}
```

I don't find any syntax error in package.json
I am trying from a long time but the same error exist. You will notice i have given absolute path for the entry as 

entry: 'main.ts',

when i give relative path **(entry: './main')**  it gives error 
ERROR in Entry module not found: Error: Cannot resolve 'file' or 'directory' ./main in D:\ProjectPath

Then it resolves the path for main.ts and finally gives the error as mentioned above.
**ERROR in Entry module not found: SyntaxError: D:\ProjectPath\package.json (directory description file): SyntaxError: Unexpected token**

This is very strange behavior. Please let me know if anybody found same issue and how did u resolve?

Thank you.

**Comment by TriMoon** [comment]
did you try ``entry: './main.ts'`` ? :wink:

**Comment by debasispanda** [comment]
Yes i have tried that ways also but no solution. It giving same error as 
SyntaxError: D:\ProjectPath\package.json 

For more information i am using bower also for some other packages. Webpack is working excellent when i am removing the package.json from the directory I am using visual studio.

**Comment by sokra** [comment]
> I am using visual studio 2013.

Does the file contains a utf-8 BOM?

**Comment by debasispanda** [comment]
The package file ? I have added the content of package.json above. Can u please suggest ?

**Comment by sokra** [comment]
> UTF-8 with signature

That's a BOM. We need to strip the DOM when loading the file while resolving...

**Comment by debasispanda** [comment]
Thanks. I just made a temporary fix by changing **Unicode(UTF-8 with signature) - Codepage 65001 encoding** to **Western European(Windows)** with the Advanced save options available in visual studio. Its working fine now.

Are we working on this for permanent solution ?

**Comment by bebraw** [comment]
Can you try against webpack 2?

**Comment by vlodko** [comment]
Having same issue when using yarn instead of npm.

**Comment by Vertygo** [comment]
It happened to me when I was pasting some package defs from web into package.json. Changing encoding to UTF-8 solved issue.

**Comment by vlodko** [comment]
@Vertygo in my case removing BOM solved the issue. Thank you for suggestion!

**Comment by blackcloud2333** [comment]
here are some problems what i had, i got the same problem and i try to solve this problem with the method above .but it can't  solve the  problem.
here is my code of package.json and i had try to convert it to UTF-8 of DOM
`"name": "react-hello-world",
  "version": "1.0.0",
  "description": "",
  "main": "main.js",
  "dependencies": {
    "ansi-styles": "^2.2.1",
    "babel-helper-builder-react-jsx": "^6.24.1",
    "babel-helper-call-delegate": "^6.24.1",
    "ansi-regex": "^2.1.1",
    "babel-helper-explode-assignable-expression": "^6.24.1",
    "babel-helper-get-function-arity": "^6.24.1",
    "babel-helper-function-name": "^6.24.1",
    "babel-helper-hoist-variables": "^6.24.1",
    "babel-helper-optimise-call-expression": "^6.24.1",
    "babel-helper-define-map": "^6.24.1",
    "babel-helper-builder-binary-assignment-operator-visitor": "^6.24.1",
    "babel-core": "^6.24.1",
    "babel-helper-regex": "^6.24.1",
    "babel-helper-remap-async-to-generator": "^6.24.1",
    "babel-helpers": "^6.24.1",
    "babel-code-frame": "^6.22.0",
    "babel-generator": "^6.24.1",
    "babel-plugin-syntax-async-functions": "^6.13.0",
    "babel-loader": "^7.0.0",
    "babel-plugin-syntax-jsx": "^6.18.0",
    "babel-messages": "^6.23.0",
    "babel-plugin-syntax-flow": "^6.18.0",
    "babel-plugin-syntax-trailing-function-commas": "^6.22.0",
    "babel-plugin-check-es2015-constants": "^6.22.0",
    "babel-plugin-syntax-exponentiation-operator": "^6.13.0",
    "babel-plugin-transform-es2015-block-scoped-functions": "^6.22.0",
    "babel-plugin-transform-es2015-classes": "^6.24.1",
    "babel-plugin-transform-es2015-destructuring": "^6.23.0",
    "babel-plugin-transform-es2015-computed-properties": "^6.24.1",
    "babel-plugin-transform-es2015-arrow-functions": "^6.22.0",
    "babel-plugin-transform-es2015-for-of": "^6.23.0",
    "babel-plugin-transform-es2015-duplicate-keys": "^6.24.1",
    "babel-plugin-transform-es2015-function-name": "^6.24.1",
    "babel-plugin-transform-es2015-literals": "^6.22.0",
    "babel-plugin-transform-es2015-block-scoping": "^6.24.1",
    "babel-plugin-transform-es2015-modules-amd": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.24.1",
    "babel-plugin-transform-es2015-object-super": "^6.24.1",
    "babel-plugin-transform-es2015-shorthand-properties": "^6.24.1",
    "babel-helper-replace-supers": "^6.24.1",
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-es2015-sticky-regex": "^6.24.1",
    "babel-plugin-transform-es2015-spread": "^6.22.0",
    "babel-plugin-transform-es2015-parameters": "^6.24.1",
    "babel-plugin-transform-exponentiation-operator": "^6.24.1",
    "babel-plugin-transform-es2015-template-literals": "^6.22.0",
    "babel-plugin-transform-es2015-unicode-regex": "^6.24.1",
    "babel-plugin-transform-react-display-name": "^6.23.0",
    "babel-plugin-transform-es2015-typeof-symbol": "^6.23.0",
    "babel-plugin-transform-react-jsx": "^6.24.1",
    "babel-plugin-transform-react-jsx-source": "^6.22.0",
    "babel-preset-es2015": "^6.24.1",
    "babel-plugin-transform-strict-mode": "^6.24.1",
    "babel-preset-es2016": "^6.24.1",
    "babel-preset-es2017": "^6.24.1",
    "babel-preset-flow": "^6.23.0",
    "babel-plugin-transform-regenerator": "^6.24.1",
    "babel-preset-latest": "^6.24.1",
    "babel-plugin-transform-react-jsx-self": "^6.22.0",
    "babel-runtime": "^6.23.0",
    "babel-template": "^6.24.1",
    "babel-types": "^6.24.1",
    "babel-preset-react": "^6.24.1",
    "babylon": "^6.17.1",
    "babel-traverse": "^6.24.1",
    "babel-plugin-transform-es2015-modules-umd": "^6.24.1",
    "balanced-match": "^0.4.2",
    "brace-expansion": "^1.1.7",
    "babel-register": "^6.24.1",
    "babel-plugin-transform-es2015-modules-systemjs": "^6.24.1",
    "concat-map": "^0.0.1",
    "commondir": "^1.0.1",
    "chalk": "^1.1.3",
    "detect-indent": "^4.0.0",
    "big.js": "^3.1.3",
    "emojis-list": "^2.1.0",
    "escape-string-regexp": "^1.0.5",
    "convert-source-map": "^1.5.0",
    "find-up": "^1.1.2",
    "find-cache-dir": "^0.1.1",
    "babel-plugin-transform-flow-strip-types": "^6.22.0",
    "debug": "^2.6.6",
    "has-ansi": "^2.0.0",
    "home-or-tmp": "^2.0.0",
    "globals": "^9.17.0",
    "is-finite": "^1.0.2",
    "invariant": "^2.2.2",
    "jsesc": "^1.3.0",
    "loadash": "^0.0.1",
    "json5": "^0.5.1",
    "minimist": "^0.0.8",
    "loader-utils": "^1.1.0",
    "minimatch": "^3.0.4",
    "mkdirp": "^0.5.1",
    "esutils": "^2.0.2",
    "os-tmpdir": "^1.0.2",
    "path-exists": "^2.1.0",
    "ms": "^0.7.3",
    "lodash": "^4.17.4",
    "pinkie-promise": "^2.0.1",
    "os-homedir": "^1.0.2",
    "pinkie": "^2.0.4",
    "pkg-dir": "^1.0.0",
    "private": "^0.1.7",
    "number-is-nan": "^1.0.1",
    "regenerator-runtime": "^0.10.5",
    "regexpu-core": "^2.0.0",
    "react": "^15.5.4",
    "regjsparser": "^0.1.5",
    "regenerator-transform": "^0.9.11",
    "path-is-absolute": "^1.0.1",
    "slash": "^1.0.0",
    "supports-color": "^2.0.0",
    "strip-ansi": "^3.0.1",
    "to-fast-properties": "^1.0.3",
    "trim-right": "^1.0.1",
    "regenerate": "^1.3.2",
    "react-dom": "^15.5.4",
    "regjsgen": "^0.2.0",
    "repeating": "^2.0.1",
    "source-map": "^0.5.6",
    "source-map-support": "^0.4.15"
  },
  "devDependencies": {},
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
     "start": "webpack-dev-server"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}
`
here is the error.
`$ webpack
Hash: 94c3df302d8dd2990ab8
Version: webpack 2.5.1
Time: 25ms

ERROR in Entry module not found: SyntaxError: E:\react-hello-world\package.json (directory description file): SyntaxError: Unexpected token ﻿ in JSON at position 0
`
how to solve this problem ?please tell me .

**Comment by Vertygo** [comment]
@blackcloud2333 not sure what tool have you used for converting this but I have used [notepad++](https://notepad-plus-plus.org/) "Encoding"->"Encode in UTF8" option from main menu.

**Comment by blackcloud2333** [comment]
yes ,i had try this method use the notepad++,besides,i also try this method in the sublime,i change the sublime with the " Reopen with encoding". 
so, i'am confuse of this now

**Comment by Vertygo** [comment]
@blackcloud2333 try to:
1. copy content of package.json (skip the first line since from the error message I would say there is some special character hidden in the beginning of the file)
2. remove package.json
3. create new file and paste content (retype first line)

**Comment by blackcloud2333** [comment]
thank you very much . 
i created the new project  just now,but the error didn't appeared,it confused to me .
maybe there are some characters what  i can seen with eyes  hidden in the package.json.

**Comment by Vertygo** [comment]
@blackcloud2333 Unfortunately you can only see those special characters if you use tools like hex editor (available in notepad++ as plugin also)

**Comment by Daddelbob** [comment]
In my case it was a extra trailing comma after the last json entry where a valid .json doesn't expect one.
![image](https://user-images.githubusercontent.com/16285370/68437125-520aa780-01c0-11ea-942f-d5bbf208b980.png)
